### PR TITLE
Features/7674 - Debug swagger issues

### PR DESCRIPTION
**Pull request by sslaws**
This change is really about reducing the time required to diagnose issues behind an lbr.  I've created scripts to locally build containers and then run them behind Traefik.  Doing this I've been able to reproduce the Swagger issue.  With the exception of the points below, everything should behave as it has:

- I've removed the IPs for Known proxies as I don't think its ever worked.
- I've removed the redirect code as we will leave that to the route to provide.
- Introduced the logger into many parts of the Swagger code for debugging.

### CODE REVIEW

**Comment by healthgatewaybc** [comment]
Medication Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=Medicationservice_pr_235)

**Comment by healthgatewaybc** [comment]
Immunization Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=immunization_pr_235)

**Comment by healthgatewaybc** [comment]
Patient Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=patientservice_pr_235)

**Comment by healthgatewaybc** [comment]
WebClient Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=webclient_pr_235)

File: Apps/Common/src/AspNetConfiguration/StartupConfiguration.cs [hunk 1]
```diff
@@ -233,11 +233,11 @@ public void UseForwardHeaders(IApplicationBuilder app)
...
                 };
-                foreach (string ip in proxyIPs)
-                {
-                    options.KnownProxies.Add(IPAddress.Parse(ip));
-                }
 
+                // foreach (string ip in proxyIPs)
+                // {
+                //     options.KnownProxies.Add(IPAddress.Parse(ip));
+                // }
```
Comment (FuriousLlama) [inline Apps/Common/src/AspNetConfiguration/StartupConfiguration.cs]: if its not necessary lets remove commented out code

File: Apps/WebClient/src/Server/Startup.cs [hunk 2]
```diff
@@ -38,7 +38,7 @@ public class Startup
     {
         private readonly StartupConfiguration startupConfig;
         private readonly IConfiguration configuration;
-
+        private readonly ILogger<Startup> logger;
```
Comment (FuriousLlama) [inline Apps/WebClient/src/Server/Startup.cs]: nit, separation with next line

File: Apps/Common/src/AspNetConfiguration/StartupConfiguration.cs (hunk 1 above)
Comment (sslaws) [reply Apps/Common/src/AspNetConfiguration/StartupConfiguration.cs]: I haven't made the determination yet... likely will be removed but this is still major flux

File: Apps/WebClient/src/Server/Startup.cs (hunk 2 above)
Comment (sslaws) [reply Apps/WebClient/src/Server/Startup.cs]: Resolved

**Comment by healthgatewaybc** [comment]
Immunization Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=immunization_pr_235)

**Comment by healthgatewaybc** [comment]
Medication Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=Medicationservice_pr_235)

**Comment by healthgatewaybc** [comment]
Patient Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=patientservice_pr_235)

**Comment by healthgatewaybc** [comment]
WebClient Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=webclient_pr_235)

File: Apps/Common/src/AspNetConfiguration/StartupConfiguration.cs [hunk 3]
```diff
@@ -326,7 +322,8 @@ public void UseHttp(IApplicationBuilder app)
             });
 
             app.UseResponseCompression();
-            app.UseHttpsRedirection();
+
+            // app.UseHttpsRedirection();
```
Comment (FuriousLlama) [inline Apps/Common/src/AspNetConfiguration/StartupConfiguration.cs]: commented out

File: Apps/Common/src/Swagger/ConfigureSwaggerGenOptions.cs [hunk 4]
```diff
@@ -82,6 +90,9 @@ private void AddSwaggerDocumentForEachDiscoveredApiVersion(SwaggerGenOptions opt
                 }
 
                 options.SwaggerDoc(description.GroupName, this.settings.Info);
+
+                // Experimenting with the document filter - this code may be removed if it doesn't pan out.
+                // options.DocumentFilter<SwaggerPathPrefixDocumentFilter>("api/medicationservice");
```
Comment (FuriousLlama) [inline Apps/Common/src/Swagger/ConfigureSwaggerGenOptions.cs]: I guess this didnt work out

**Pull request merged**